src/relay/pass/type_solver.cc

```diff
@@ -516,17 +638,21 @@ TVM_REGISTER_API("relay._analysis._test_type_solver")
 .set_body([](runtime::TVMArgs args, runtime::TVMRetValue* ret) {
     using runtime::PackedFunc;
     using runtime::TypedPackedFunc;
-    ErrorReporter err_reporter;
-    auto solver = std::make_shared<TypeSolver>(GlobalVarNode::make("test"), &err_reporter);
+    ErrorReporter *err_reporter = new ErrorReporter();
+    auto solver = std::make_shared<TypeSolver>(GlobalVarNode::make("test"), err_reporter);
 
-    auto mod = [solver](std::string name) -> PackedFunc {
+    auto mod = [solver, err_reporter](std::string name) -> PackedFunc {
```

Comment:
BAD! This will never get free!
use a shared_ptr!

Reply:
Its fine, this is testing code dude